src/framework/theme/components/icon/icon-libraries.ts

```diff
@@ -0,0 +1,184 @@
+    this.packs.set(name, {
+      name,
+      icons: new Map(Object.entries(icons)),
+      params,
+      type: NbIconPackType.SVG,
+    });
+  }
+
+  /**
+   * Registers new font pack
+   * @param {string} name
+   * @param {NbIconPackParams} params
+   */
+  registerFontPack(name: string, params: NbIconPackParams = { packClass: '' }) {
+    this.packs.set(name, {
+      name,
+      params,
+      icons: new Map(),
+      type: NbIconPackType.FONT,
+    });
+  }
+
+  /**
+   * Returns pack by name
+   * @param {string} name
+   */
+  getPack(name: string) {
+    return this.packs.get(name);
+  }
+
+  /**
+   * Sets pack as a default
+   * @param {string} name
+   */
+  setDefaultPack(name: string) {
+    if (!this.packs.has(name)) {
+      throwPackNotFoundError(name);
+    }
+
+    this.defaultPack = this.packs.get(name);
+  }
+
+  /**
+   * Returns Svg icon
+   * @param {string} name
+   * @param {string} pack
+   *
+   * @returns NbIconDefinition
+   */
+  getSvgIcon(name: string, pack?: string): NbIconDefinition {
+
+    const iconsPack = this.getPackOrDefault(pack);
+
+    if (iconsPack.type !== NbIconPackType.SVG) {
+      throw Error(`Pack '${iconsPack.name}' is not an SVG Pack and its type is '${iconsPack.type}'`);
+    }
+
+    const icon = this.getIconFromPack(name, iconsPack);
+
+    return {
+      name,
+      pack: iconsPack.name,
+      type: NbIconPackType.SVG,
+      icon: this.createSvgIcon(name, icon, iconsPack.params),
+    };
+  }
+
+  /**
+   * Returns Font icon
+   * @param {string} name
+   * @param {string} pack
+   *
+   * @returns NbIconDefinition
+   */
+  getFontIcon(name: string, pack?: string): NbIconDefinition {
+    const iconsPack = this.getPackOrDefault(pack);
+
+    if (iconsPack.type !== NbIconPackType.FONT) {
+      throw Error(`Pack '${iconsPack.name}' is not a Font Pack and its type is '${iconsPack.type}'`);
+    }
+
+    const icon = this.getIconFromPack(name, iconsPack, false);
+
+    return {
+      name,
+      pack: iconsPack.name,
+      type: NbIconPackType.FONT,
+      icon: this.createFontIcon(name, icon ? icon : '', iconsPack.params),
+    };
+  }
+
+  /**
+   * Returns an icon
+   * @param {string} name
+   * @param {string} pack
+   *
+   * @returns NbIconDefinition
+   */
+  getIcon(name: string, pack?: string) {
+
+    const iconsPack = this.getPackOrDefault(pack);
+
+    if (iconsPack.type === NbIconPackType.SVG) {
+      return this.getSvgIcon(name, pack);
+    }
+
+    return this.getFontIcon(name, pack);
+  }
+
+  protected createSvgIcon(name: string, content: NbIcon | string, params: NbIconPackParams) {
+    return content instanceof NbSvgIcon ? content : new NbSvgIcon(name, content, params);
+  }
+
+  protected createFontIcon(name: string, content: NbIcon | string, params: NbIconPackParams) {
+    return content instanceof NbFontIcon ? content : new NbFontIcon(name, content, params);
+  }
+
+  protected getPackOrDefault(name: string) {
```

Comment:
Method name is confusing. Based on the name I'll expect method to return pack with `name` or default pack if pack with `name` not found.
This method will throw if icon pack with a given name not found regardless of default pack.
If this is expected behavior, then method needs different name.
It's also hard to unravel when this method will throw. I'd prefer something more linear.
```
if (name) {
  const pack: NbIconPack = this.packs.get(name);

  if (pack) {
    return pack;
  } else {
    throwPackNotFoundError(name);
  }
}

if (this.defaultPack) {
  return this.defaultPack;
}
throwNoDefaultPackError();
```

And the last one, part of service methods lack return type. I think methods with specified return types way more easier to read, instead of looking into realization.